lib/foreman_webhooks/engine.rb

```diff
@@ -18,24 +18,46 @@ class Engine < ::Rails::Engine
+          permission :view_webhooks,    { webhooks: %i[index show],
+                                          'api/v2/webhooks': %i[index show] }, resource_type: 'Webhook'
+          permission :create_webhooks,  { webhooks: %i[new create],
+                                          'api/v2/webhooks': [:create] }, resource_type: 'Webhook'
+          permission :edit_webhooks,    { webhooks: %i[edit update],
+                                          'api/v2/webhooks': [:update] }, resource_type: 'Webhook'
+          permission :destroy_webhooks, { webhooks: [:destroy],
+                                          'api/v2/webhooks': [:destroy] }, resource_type: 'Webhook'
+          permission :view_payload_templates,     { payload_templates: [:index, :show, :auto_complete_search, :preview, :export],
+                                                    'api/v2/payload_templates': [:index, :show, :export] },
+                                                  resource_type: 'PayloadTemplate'
+          permission :create_payload_templates,   { payload_templates: [:new, :create, :clone_template],
+                                                    'api/v2/payload_templates': [:create, :clone, :import] },
+                                                  resource_type: 'PayloadTemplate'
+          permission :edit_payload_templates,     { payload_templates: [:edit, :update],
+                                                    'api/v2/payload_templates': [:update, :import] },
+                                                  resource_type: 'PayloadTemplate'
+          permission :destroy_payload_templates,  { payload_templates: [:destroy],
+                                                    'api/v2/payload_templates': [:destroy] },
+                                                  resource_type: 'PayloadTemplate'
+          permission :lock_payload_templates,     { payload_templates: [:lock, :unlock],
+                                                    'api/v2/payload_templates': [:lock, :unlock] },
+                                                  resource_type: 'PayloadTemplate'
         end
 
         # add menu entry
-        menu :admin_menu, :webhook_targets, url_hash: { controller: :webhook_targets, action: :index },
-                                            caption: N_('Webhook Targets'),
-                                            parent: :administer_menu
-
+        divider :admin_menu, caption: N_('Webhook'), parent: :administer_menu
+        menu :admin_menu, :webhooks, url_hash: { controller: :webhooks, action: :index },
+                                     caption: N_('Webhooks'),
+                                     parent: :administer_menu
+        menu :admin_menu, :payload_templates, url_hash: { controller: :payload_templates, action: :index },
+                                              caption: N_('Payload Templates'),
```

Comment:
To be more consistent with the item "Webhooks" above, I suggest to rename this to "Webhook Templates". Payload is too generic term.